### PR TITLE
Query multiple tables

### PR DESCRIPTION
**Pull request by Apatakiti**
### Project requirements
- Create a table named owners with the following columns:

- id: integer (set it as autoincremented PRIMARY KEY)
- full_name: string
- age: integer
- 
- ##### Create a table named species with the following columns:

- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- Modify animals table:
- Make sure that id is set as autoincremented PRIMARY KEY
- Remove column species
- Add column species_id which is a foreign key referencing species table
- Add column owner_id which is a foreign key referencing the owners table

- id: integer (set it as autoincremented PRIMARY KEY)
- name: string
- Modify animals table:
- Make sure that id is set as autoincremented PRIMARY KEY
- Remove column species

- Add column species_id which is a foreign key referencing species table
- Add column owner_id which is a foreign key referencing the owners table
- Remember all this goes in schema.sql file.
##### - Insert the following data into the owners table:
- Sam Smith 34 years old.
- Jennifer Orwell 19 years old.
- Bob 45 years old.
- Melody Pond 77 years old.
- Dean Winchester 14 years old.
- Jodie Whittaker 38 years old.
- Insert the following data into the species table:
- Pokemon
- Digimon
##### - Modify your inserted animals so it includes the species_id value:
- If the name ends in "mon" it will be Digimon
- All other animals are Pokemon
##### - Modify your inserted animals to include owner information (owner_id):
- Sam Smith owns Agumon.
- Jennifer Orwell owns Gabumon and Pikachu.
- Bob owns Devimon and Plantmon.
- Melody Pond owns Charmander, Squirtle, and Blossom.
- Dean Winchester owns Angemon and Boarmon.
- Remember these insertions and modifications happen in data.sql file.
##### - Write queries (using JOIN) to answer the following questions:
- What animals belong to Melody Pond?
- List of all animals that are pokemon (their type is Pokemon).
- List all owners and their animals, remember to include those that don't own any animal.
- How many animals are there per species?
- List all Digimon owned by Jennifer Orwell.
- List all animals owned by Dean Winchester that haven't tried to escape.
- Who owns the most animals?
- Remember all these should be written in queries.sql file.
##### Remember to add the screenshot with the results of your queries to your Pull Request description.

Screenshot (37)

Screenshot (41)

Screenshot (42)

Screenshot (43)

Screenshot (44)

Screenshot (45)

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -68,3 +70,25 @@ WHERE date_of_birth
 BETWEEN '1990-01-01' 
 AND '2000-12-31' 
 GROUP BY species;
+
+-- Project 3
+--Join Query by owner name
+select * from animals JOIN owners on animals.owner_id=owners.id where owners.full_name='Melody Pond';
+
+--Join Query by species name
+select * from animals JOIN species on animals.species_id=species.id where species.name='Pokemon';
```
Comment (franclobo) [inline queries.sql]: - [ ] Please 🙏🏽 correct the print screen in your PR, it is supposed to show only the Pokemon list.

File: queries.sql [hunk 2]
```diff
@@ -68,3 +70,25 @@ WHERE date_of_birth
...
+
+--Full Join Query by owner
+select * from animals Full JOIN owners on animals.owner_id = owners.id;
```
Comment (franclobo) [inline queries.sql:94]: - [ ] Kindly remember to include those who don't own any animal, this is a requirement 👇🏽 

> List all owners and their animals, remember to include those that don't own any animal.

File: queries.sql [hunk 3]
```diff
@@ -68,3 +70,25 @@ WHERE date_of_birth
...
+
+--Join Count group by species
+select species.name as animals_species,count(species.name) as species_count from species JOIN animals on animals.species_id=species.id GROUP BY species.name;
```
Comment (franclobo) [inline queries.sql]: - [ ] Please show the number of pokemon (4) and digimon (6), not a list of them. That's the requirement 👇🏽 

> How many animals are there per species?

File: queries.sql [hunk 4]
```diff
@@ -68,3 +70,25 @@ WHERE date_of_birth
...
+
+--Join three tables and filter by owner and species
+select * from animals JOIN owners on animals.owner_id=owners.id JOIN species on animals.species_id=species.id where species.name='Digimon' and owners.full_name='Jennifer Orwell';
+
+--Join animal and owner table and filter by owner name and zero escape attempt
+select * from animals JOIN owners on animals.owner_id=owners.id WHERE owners.full_name='Dean Winchester' AND escape_attempts=0;
+
+-- Who owns the most animals;
+select full_name as owner_name, count(full_name)no_of_animals from owners JOIN animals on animals.owner_id=owners.id GROUP BY owners.full_name;
```
Comment (franclobo) [inline queries.sql:103]: - [ ] I couldn't find the correct answer for this queries, please 🙏🏽 correct the print screens and queries to have the correct information of the requirements. 👇🏽 

`Sam Smith` 1
`Jennifer Orwell` 2
`Bob` 2
`Melody Pond` 3
`Dean Winchester` 2

List all Digimon owned by Jennifer Orwell. = `Gabumon`
List all animals owned by Dean Winchester that haven't tried to escape.
Who owns the most animals? = `Melody Pond` 3

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @Ademola-coding, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Apatakiti) [reply queries.sql]: corrected

Screenshot (54)

File: queries.sql (hunk 2 above)
Comment (Apatakiti) [reply queries.sql:94]: corrected

Screenshot (55)

File: queries.sql (hunk 3 above)
Comment (Apatakiti) [reply queries.sql]: corrected

Screenshot (55)

File: queries.sql (hunk 4 above)
Comment (Apatakiti) [reply queries.sql:103]: some data were inserted twice that's the reason some output data were altered
Screenshot (56)

Screenshot (57)

some data were inserted twice that's the reason some output data were altered

Screenshot (59)

**Review by Alusp — APPROVED** [review]
Hi @Ademola-coding, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats
## Highlights

- [x] Use primary key & foreign key mechanism for joining tables.👍
- [x] Prepare complex queries that answer analytical questions. 👍
- [x] Use Professional Readme file 👍
- [x] Descriptive PR Summary used 👍

## Optional suggestions

- [OPTIONAL BUT HIGHLY RECOMMENDED] Please ensure that you put your Screenshots in your PR description instead of the comment sections. I can see that your screenshots in the comment sections are correct but not ordered.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me@Alusp in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**